packages/framework/fluid-static/src/rootDataObject.ts

```diff
@@ -125,8 +125,11 @@ export class RootDataObject extends DataObject<{ InitialState: RootDataObjectPro
     private async createDataObject<T extends IFluidLoadable>(dataObjectClass: DataObjectClass<T>): Promise<T> {
         const factory = dataObjectClass.factory;
         const packagePath = [...this.context.packagePath, factory.type];
-        const router = await this.context.containerRuntime.createDataStore(packagePath);
-        return requestFluidObject<T>(router, "/");
+        const dataStore = await this.context.containerRuntime.createDataStore(packagePath);
+        const maybeIFluidLoadable: IDataStore & FluidObject<IProvideFluidLoadable> = dataStore;
+        const entrypoint = maybeIFluidLoadable.IFluidLoadable?.handle?.get();
+        assert(entrypoint instanceof DataObject, "The data store's handle is not of the correct type!");
```

Comment:
maybe add a comment why this is safe

Reply:
Is this actually safe?  It seems not safe if e.g. the schema includes a DataObject that was built using Aqueduct version N-1?

Reply:
I spoke with @ChumpChief and it feels safer to just drop the assert for now, given that there was no such thing in the current version of the code. @anthony-murphy I can update you when you're back about my chat with Matt and we can schedule follow-up work to address this if necessary.